apicast/src/provider.lua

```diff
@@ -385,8 +385,15 @@ function _M.access(service)
     ngx.var.cached_key = concat({service.id, params.app_id, params.app_key}, ':')
 
   elseif backend_version == 'oauth' then
-    ngx.var.access_token = parameters.access_token
-    params.access_token = parameters.access_token
+    if credentials.location == 'headers' then
```

Comment:
Hmm, not sure about it. Other auth modes accept credentials **either** in querystring/body **or** in headers, and there's a switch in UI for that.

I think for OAuth it would make sense to check both locations, and use whatever is found. But this would make the switch in the UI useless...